### PR TITLE
fix: nuxt terminal error

### PR DESCRIPTION
**Pull request by csandru-plenty**
## Why:

Closes: [AB#126530](https://dev.azure.com/plentymarkets/0af57df9-8662-4901-bb97-8e88b07ff0c9/_workitems/edit/126530)

## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: apps/web/components/PayPal/PayPalExpressButton.vue [hunk 1]
```diff
@@ -40,9 +37,8 @@ const checkOnClickEvent = (): boolean => {
 };
 
 const onInit = (actions: OnInitActions) => {
-  if (type.value === TypeCheckout) {
-    disabled.value ? actions.disable() : actions.enable();
-    watch(disabled, () => (disabled.value ? actions.disable() : actions.enable()));
```
Comment (maxiroellplenty) [inline apps/web/components/PayPal/PayPalExpressButton.vue:45]: why was the watch removed?

File: apps/web/pages/[...slug].vue [hunk 2]
```diff
@@ -40,8 +39,8 @@ const { data: categoryTree } = useCategoryTree();
 const { buildCategoryLanguagePath } = useLocalization();
 
 const handleQueryUpdate = async () => {
-  await fetchProducts(getFacetsFromURL());
-  checkFiltersInURL();
+  await fetchProducts(getFacetsFromURL()).then(() => checkFiltersInURL());
```
Comment (maxiroellplenty) [inline apps/web/pages/[...slug].vue:42]: what is the difference? shouldn't it work the same with the await and next line?

File: apps/web/pages/[...slug].vue [hunk 3]
```diff
@@ -50,24 +49,28 @@ const handleQueryUpdate = async () => {
   }
 };
 
-await handleQueryUpdate();
-
-const category = ref(productsCatalog.value.category);
+onNuxtReady(async () => {
+  await handleQueryUpdate().then(() => setCategoriesPageMeta(productsCatalog.value, getFacetsFromURL()));
```
Comment (maxiroellplenty) [inline apps/web/pages/[...slug].vue:53]: what is the difference? shouldn't it work the same with the await and next line?

File: apps/web/pages/[...slug].vue [hunk 4]
```diff
@@ -50,24 +49,28 @@ const handleQueryUpdate = async () => {
...
+  }
+
+  return [];
 });
 
 watch(
   () => locale.value,
   async (changedLocale: any) => {
-    router.push({
+    await router.push({
```
Comment (maxiroellplenty) [inline apps/web/pages/[...slug].vue]: why do we need the await here. Also from what I know a watch that is async has to be saved into a variable and the watch has to be destroyed when the component unmounts. https://vuejs.org/guide/essentials/watchers.html#stopping-a-watcher

File: apps/web/pages/[...slug].vue [hunk 5]
```diff
@@ -77,23 +80,33 @@ watch(
 watch(
   () => route.query,
   async () => {
-    await handleQueryUpdate();
+    await handleQueryUpdate().then(() => setCategoriesPageMeta(productsCatalog.value, getFacetsFromURL()));
```
Comment (maxiroellplenty) [inline apps/web/pages/[...slug].vue:83]: that makes more sense

**Review by maxiroellplenty — CHANGES_REQUESTED** [review]

File: apps/web/components/PayPal/PayPalExpressButton.vue (hunk 1 above)
Comment (csandru-plenty) [reply apps/web/components/PayPal/PayPalExpressButton.vue:45]: 1. we are inside a method
2. method call
3. trigger disabled.value ? actions.disable() : actions.enable();
4. watcher inside a method (!), watcher doing the same thing as point 3

File: apps/web/pages/[...slug].vue (hunk 2 above)
Comment (csandru-plenty) [reply apps/web/pages/[...slug].vue:42]: both are similar in that they execute checkFiltersInURL() after fetchProducts(getFacetsFromURL()), they are not exactly the same in terms of how they handle errors.

Initial version:
checkFiltersInURL() is called after fetchProducts(getFacetsFromURL()) completes successfully. If fetchProducts throws an error, the execution will stop, and checkFiltersInURL() will not be called
This version:
If fetchProducts resolves successfully, checkFiltersInURL() will be called. However, if fetchProducts throws an error, checkFiltersInURL() will not be called, similar to the first version. However, this version allows for chaining and could potentially lead to handling the error in a .catch(), if we choose to add one; isolates the scope of eventual errors, making it easier to debug.

File: apps/web/pages/[...slug].vue (hunk 3 above)
Comment (csandru-plenty) [reply apps/web/pages/[...slug].vue:53]: more or less, same principle as the above explanation: enforces successful chain completion

File: apps/web/pages/[...slug].vue (hunk 4 above)
Comment (csandru-plenty) [reply apps/web/pages/[...slug].vue]: Removed (had the async there from previous version and IDE auto fix added await).
On async topic:
should be used if we need to wait for the navigation to complete before executing subsequent code. For example, if you want to perform an action after the navigation is confirmed.
On watcher destroy/not:
as it says on your shared url, "are bound to the owner component instance, and will be automatically stopped when the owner component is unmounted". nothing to worry about, can be easily tested introducing a console.log into it I suppose...

**Review by maxiroellplenty — APPROVED** [review]

File: apps/web/components/PayPal/PayPalExpressButton.vue (hunk 1 above)
Comment (maxiroellplenty) [reply apps/web/components/PayPal/PayPalExpressButton.vue:45]: got it 👍

File: apps/web/pages/[...slug].vue (hunk 2 above)
Comment (maxiroellplenty) [reply apps/web/pages/[...slug].vue:42]: 👍

File: apps/web/pages/[...slug].vue (hunk 4 above)
Comment (maxiroellplenty) [reply apps/web/pages/[...slug].vue]: ok if thats the case then there is nothing to worry about

**Pull request merged**